helm/cluster-vsphere/templates/vspheremachinetemplate.yaml

```diff
@@ -17,7 +17,7 @@ spec:
 apiVersion: {{ include "infrastructureApiVersion" . }}
 kind: VSphereMachineTemplate
 metadata:
-  name: {{ include "resource.default.name" $ }}-{{ $nodePoolName }}-{{ include "mtRevision" $c }}
+  name: {{ include "resource.default.name" $ }}-{{ $nodePoolName }}-{{ include "machineTemplateSpec.hash" (dict "data" (include "worker-vspheremachinetemplate-spec" $) "salt" $.Values.cluster.providerIntegration.hashSalt) }}
```

Comment:
the naming function that ensures machines have the same names as referenced inside the cluster chart resources